tests/components/honeywell/test_config_flow.py

```diff
@@ -36,34 +37,35 @@ async def test_show_authenticate_form(hass: HomeAssistant) -> None:
 async def test_connection_error(hass: HomeAssistant, client: MagicMock) -> None:
     """Test that an error message is shown on connection fail."""
     client.login.side_effect = AIOSomecomfort.ConnectionError
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": SOURCE_USER}, data=FAKE_CONFIG
-    )
+    with patch("AIOSomecomfort.AIOSomeComfort", return_value=client):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=FAKE_CONFIG
+        )
     assert result["errors"] == {"base": "cannot_connect"}
 
 
 async def test_auth_error(hass: HomeAssistant, client: MagicMock) -> None:
     """Test that an error message is shown on login fail."""
     client.login.side_effect = AIOSomecomfort.AuthError
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": SOURCE_USER}, data=FAKE_CONFIG
-    )
+    with patch("AIOSomecomfort.AIOSomeComfort", return_value=client):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=FAKE_CONFIG
+        )
     assert result["errors"] == {"base": "invalid_auth"}
 
 
-async def test_create_entry(hass: HomeAssistant) -> None:
+async def test_create_entry(hass: HomeAssistant, client: MagicMock) -> None:
     """Test that the config entry is created."""
-
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": SOURCE_USER}, data=FAKE_CONFIG
-    )
+    with patch("AIOSomecomfort.AIOSomeComfort", return_value=client):
```

Comment:
Please also patch the integration setup function here to avoid setting up the whole integration. Ie patch `async_setup_entry`.

Reply:
I guess I new that - sorry about the extra round.